vault/logical_system_custom_messages_test.go

```diff
@@ -437,6 +437,155 @@ func TestHandleCreateCustomMessage(t *testing.T) {
 	assert.Contains(t, resp.Data, "error")
 }
 
+// TestHandleCreateAndUpdateCustomMessageLinkProperty verifies some particular
+// cases in the handleCreateCustomMessage method surrounding the link property
+// of the custom message. Because the link property is modeled as a JSON object,
+// the possibility of an empty or partially constructed object is provided, so
+// this test function verifies that each of these cases is appropriately
+// handled.
+func TestHandleCreateAndUpdateCustomMessageLinkProperty(t *testing.T) {
+	testBackend := &SystemBackend{
+		Core: &Core{
+			customMessageManager: uicustommessages.NewManager(&logical.InmemStorage{}),
+		},
+	}
+
+	fieldData := &framework.FieldData{
+		Schema: map[string]*framework.FieldSchema{
+			"title": {
+				Type:     framework.TypeString,
+				Required: true,
+			},
+			"message": {
+				Type:     framework.TypeString,
+				Required: true,
+			},
+			"authenticated": {
+				Type:    framework.TypeBool,
+				Default: true,
+			},
+			"type": {
+				Type:    framework.TypeString,
+				Default: "banner",
+			},
+			"start_time": {
+				Type:     framework.TypeTime,
+				Required: true,
+			},
+			"end_time": {
+				Type: framework.TypeTime,
+			},
+			"link": {
+				Type: framework.TypeMap,
+			},
+			"options": {
+				Type: framework.TypeMap,
+			},
+		},
+		Raw: map[string]any{
+			"title":      "The title",
+			"message":    "VGhlIG1lc3NhZ2UK",
+			"start_time": "2024-01-01T00:00:00.000Z",
+		},
+	}
+
+	for _, testcase := range []struct {
+		name                  string
+		linkField             any
+		responseErrorExpected bool
+	}{
+		{
+			name:      "nil-link",
+			linkField: nil,
+		},
+		{
+			name:      "empty-link-map",
+			linkField: map[string]any{},
+		},
+		{
+			name: "empty-key-and-value-link-map",
```

Comment:
How come these don't have `responseErrorExpected: true`?

Reply:
This is a case where the request body contains:
```
{
  ...
 
 "link": {"": ""},
  ...
}
```

In this particular case, the thought is that there's no link information (partial or invalid), so it should be treated as though **link** was not included in the request body. Thus, no error response is generated, instead the message is created/updated to not have a link.

Reply:
All remaining test cases fall in the same case. I could add a comment above each group of test cases to further explain the common behaviour with each group of testcases.